nocodec-defconfig

```diff
@@ -1,2 +1,3 @@
+CONFIG_SND_SOC_SOF_HDA_LINK=n
 CONFIG_SND_SOC_SOF_NOCODEC=m
 # CONFIG_SND_SOC_SOF_FORCE_NOCODEC_MODE is not set
```

Comment:
I suggest to enable force no codec.
For some device that had ACPI in BIOS, we will try to probe some machine driver that is not existed.
Even if you selected the machine driver. It will still be no possible to probe success due to lack support for HDMI audio.

Reply:
NO. if you use nocodec, then don't use any other machine drivers. that's the whole point about this PR.

Reply:
@plbossart 
The problem here is that if you did not select the machine driver. But the SND_SOC_ACPI_INTEL_MATCH is always enabled. If the acpi id is there, we will try to probe a machine driver that is not existed.

Reply:
Ah yes, you are right. I was thinking that it's useless e.g. for MinnowBoard and Up2 since you can remove the ACPI stuff from the SSDT, but on generic machines that wouldn't work.
Will fix, good feedback indeed.

Reply:
yes, that is my point. As we have more device with codec on board. It is impossible to delete the ACPI.